02-KernelModules_CharacterDevice/device_creation.h

```diff
@@ -0,0 +1,97 @@
+#ifndef CREATE_DEV
+#define CREATE_DEV
+#include <linux/fs.h> //for getting free MAJOR and MINOR
+#include <linux/cdev.h> //char_dev creation
+#include <linux/types.h> //for dev_t
+#include <linux/uaccess.h> //copy_to/from_user
+#include <linux/device.h> //for device_creaete
+#include "Memory_management.h" //driver_buffer, read_position, data_length
+#define NUMBER_OF_DEVICES   1
+#define DEVICE_NAME "My_character_device"
+
+ssize_t dev_read (struct file *file_p, char __user *usr_buffer, size_t len, loff_t *offset);
+ssize_t dev_write (struct file *file_p, const char __user *usr_buffer, size_t len, loff_t *offset);
+
+
+static struct file_operations my_fops = {
```

Comment:
```suggestion

```
Remove implementation  a function . There must be a function declaration
Remove file_operation from .h